src/wodan/wodan.ml

```diff
@@ -224,19 +232,16 @@ type node = {
      A flushable node is either new or dirty, but not both. *)
   mutable flush_children : int64 KeyedMap.t option;
   (* Use offsets so that data isn't duplicated *)
-  mutable children : int64 KeyedMap.t Lazy.t;
+  mutable children : int64 KeyedMap.t;
   (* Don't reference nodes directly, always go through the
- * LRU to bump recently accessed nodes *)
-  children_alloc_ids : int64 KeyedMap.t;
+   * LRU to bump recently accessed nodes *)
+  mutable children_alloc_ids : int64 KeyedMap.t;
```

Comment:
`KeyedMap`s are already mutables, so I think we should avoid making them mutable here (see my previous comment).

Reply:
I have the opposite viewpoint: since we break with functional purity, we have no need to pretend anything is immutable here.

Reply:
(indeed, this does get mutated in non-root splits)